### ISSUE TITLE
cross-site cookies are blocked for browser extensions?

### ISSUE THREAD

**Issue by martgil** [brave/brave-browser#24459]
## Description 
The FlowCrypt browser extension doesn't work with Brave browser Shields turned on. I tried digging deeper into the Brave's Shields options and found out that it only happens when the cross-site cookies are blocked -- the extension works well by allowing the cookies. The browser extension injects an iframe and renders it on mail.google.com

I found an existing Github issue that seems related to what we've been experiencing - https://github.com/brave/brave-browser/issues/12791

## Steps to Reproduce
Download the FlowCrypt browser extension at https://chrome.google.com/webstore/detail/cryptup-encrypt-gmail-wit/bnjglocicdkmhmoohhfkfkbbkejdhdgc and set it up

   1. send an encrypted email to yourself
   2. open the encrypted email with Brave browser shields turned on (blocked cross-site cookie) 
   3. to resolve the issue temporarily, turn off brave browser shields or allow cookies on the advanced settings of Brave browser shields

## Actual result:

(blocked cross-site cookies)
![Screenshot from 2022-08-05 14-27-21](https://user-images.githubusercontent.com/46025304/183019840-db8e1474-c4c1-4716-bcda-9d8efef399e2.png)

(allow cookies)
![Screenshot from 2022-08-05 14-28-33](https://user-images.githubusercontent.com/46025304/183020061-265d440d-1701-4058-b7a9-718b4e80db69.png)

## Reproduces how often: 
Easily reproduced

## Brave version (brave://version info)
```
Brave	1.42.86 Chromium: 104.0.5112.81 (Official Build) (64-bit) 
Revision	5b7b76419d50f583022568b6764b630f6ddc9208-refs/branch-heads/5112@{#1309}
OS	Linux
JavaScript	V8 10.4.132.20
User Agent	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.5112.81 Safari/537.36
Command Line	/opt/brave.com/brave/brave --component-updater=url-source=https://go-updater.brave.com/extensions --desktop-startup-id=gnome-shell/Brave Web Browser/2767-1-martPC_TIME45953 --disable-domain-reliability --disable-features --enable-crashpad --enable-dom-distiller --enable-features --lso-url=https://no-thanks.invalid --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --variations-insecure-server-url=https://variations.brave.com/seed --variations-server-url=https://variations.brave.com/seed --enable-crashpad --enable-features --disable-features --flag-switches-begin --flag-switches-end
Executable Path	/opt/brave.com/brave/brave
Profile Path	/home/mart/.config/BraveSoftware/Brave-Browser/Default
Active Variations	AdRewardsStudy:NextPaymentDay
AdServingStudy:DefaultAdNotificationsPerHour=10/MaximumAdNotificationsPerDay=100/MaximumInlineContentAdsPerHour=12/MaximumInlineContentAdsPerDay=60/AdServingVersion=1
BraveAds.UserActivityStudy:Triggers=NTPSI/TimeWindow=15m/Threshold=0.0/IdleTimeThreshold=5s
BraveDebounceStudy:Enabled
ConversionsStudy:NewDefaultPattern
Default1pBlockingStudy:Disabled
DisableReduceLanguage:Enabled
EphemeralStorageStudy:Enabled
ExtendedVariationsSafeMode5:SignalAndWriteViaFileUtil5
FirstPartyEphemeralDomainBlockStudy:Enabled
ForceDisableBlinkFeatures:Enabled
PartitionBlinkMemoryCacheStudy:Enabled
PartitionConnectionsByNetworkIsolationKeyStudy:Enabled
PermissionLifetimeReleaseStudy:Enabled
RequestAdsEnabledApiStudy:Enabled
```
## Version/Channel Information:

- Can you reproduce this issue with the current release? yes
- Can you reproduce this issue with the beta channel? yes
- Can you reproduce this issue with the nightly channel? yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? yes
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? yes

**Comment by martgil** [comment]
We will be following any requirements, suggestions, or changes in the extension's [manifest file](https://github.com/FlowCrypt/flowcrypt-browser/blob/master/extension/manifest.json) to resolve the issue as some of the Brave browser users would like to use the extension without disabling Brave browser shields when accessing sites such as mail.google.com

**Comment by antonok-edm** [comment]
@martgil as per https://github.com/brave/brave-browser/issues/12791, this appears to be an intentional privacy enhancement. That being said, have you attempted the suggestions in https://github.com/brave/brave-browser/issues/12791#issuecomment-734059788?

> probably the best thing we can do is encourage extensions to **handle long term, global, extension stored using the extension API, or in the background script in general**.

cc @pes10k

**Comment by pes10k** [comment]
Yes this is intentional. Once something is injected in the page, chromium / blink does not provide a guaranteed-correct way of attributing the behavior to the extension or the page itself. 

As @antonok-edm mentioned, the correct behavior is for the extension to handle storage that it doesn’t want to affected by browser-imposed page restrictions to manage that storage extension side
